### ISSUE TITLE
How to tell between free text input and a suggestion selection

### ISSUE THREAD

**Issue by gravitz** [corejavascript/typeahead.js#113]
This dates back to the original typeahead. Having struggled with this for the whole day including scouring the internet I have yet to arrive at a solution to this. I was hoping typeahead:change event would provide a means to make out a selection from custom input values, but it appears to not be the case.

Has anyone been able to solve this? Alternative suggestions are welcome.

BR,

Gravitz

**Comment by vlahanas** [comment]
@gravitz  

To achieve this, i use a combination of 'change' and 'typeahead:_done' events. The first is fired whenever the input loses focus (and its value has changed), while the latter is fired only when a suggestion selection occurs. Also, 'typeahead:_done' is always fired last. 

So you can raise a free text flag within the 'change' event and "unraise" it inside the _done event.

**Comment by ViRuSTriNiTy** [comment]
@vlahanas  ...but there isnt a _done event?!?

**Comment by vlahanas** [comment]
Yes @ViRuSTriNiTy , sorry if i wasn't clear enough. There is a 'typeahead:_done' event, 

ex. 

$('#element-id).on('typeahead:_done , function(evt, object, datum) {
				// do what you want with datum here
			});
